### DISCUSSION TITLE
workaround for addDevice dbus

### DISCUSSION THREAD

**Discussion by yjeanrenaud** [AsamK/signal-cli#433] [Q&A]
it is a pitty we can't use `signal-cli --dbus addDevice --uri `
it just says.
`addDevice is not yet implemented via dbus`

is there a workaround for that?

otherwise, `link` would be as helpfull as well, cause then we could use all other features of signal-cli and leave the dbus daemon for automatisation etc.

**Comment by johnfreed** [comment]
I have been working on this issue but it's of limited value. `link` and `addDevice` normally refer to a Signal Desktop client or a Signal smartphone client, not to another instance of `signal-cli`. I have added a brief HOWTO to the wiki entry for linking to an instance of `signal-cli` on a remote machine, but I'm not sure that is what you are looking for.

Dbus is not designed to transmit large amounts of data (like the original Signal message attachments), so at this point only metadata is sent other than for the text of the message itself. (This formally limits the message to 128Mb, the maximum size of a Dbus command.) So the Dbus daemon is indeed designed just for "automatisation etc." See my sample implementation of a Signal-to-email program (https://github.com/johnfreed/signalmail).

It's theoretically possible on the primary machine to run a renamed signal-cli (signal-secondary?) that would supervise only secondary devices out of a different data directory (other than $HOME/.local/share/signal-cli) and perhaps others have already done so. I haven't tried to work on that.
